concrete/src/Express/Form/Validator/Routine/CaptchaRoutine.php

```diff
@@ -24,11 +24,16 @@ public function __construct(CaptchaInterface $service)
      */
     public function validate(ErrorList $error, Form $form, Entry $entry = null)
     {
-        if (!$this->captchaValidator->check()) {
-            $error->add(t('Incorrect captcha code.'));
-
-            return false;
-        }
+		if (method_exists($this->captchaValidator, 'checkWithErrorList')) {
```

Comment:
I'd write
if ... instanceof CaptchaWithErrorInterface

Reply:
... except then any plugins that want to use that function must explicitly implement CaptchaWithErrorInterface and won't be backward compatible. By checking method_exists rather than instanceof they only have to explicitly implement CaptchaInterface and should work on previous versions.